server/api.js

```diff
@@ -77,10 +77,53 @@ async function getSensor(req, res) {
   }
 }
 
+const validateMessageClients = Validator.body(['braveKey', 'message']).exists()
+
+async function messageClients(req, res) {
+  const validationErrors = Validator.validationResult(req).formatWith(helpers.formatExpressValidationErrors)
+
+  try {
+    if (validationErrors.isEmpty()) {
+      const clients = await db.getActiveClients()
+      const message = req.body.message
+
+      for (const c of clients) {
+        // define phone_numbers as a Set to prevent duplicate numbers
+        const phone_numbers = new Set()
+
+        // add responder, fallback, and heartbeat numbers to phone_numbers
+        c.responderPhoneNumbers.forEach(pn => {
+          phone_numbers.add(pn)
+        })
+        c.fallbackPhoneNumbers.forEach(pn => {
+          phone_numbers.add(pn)
+        })
+        c.heartbeatPhoneNumbers.forEach(pn => {
+          phone_numbers.add(pn)
+        })
```

Comment:
That is actually a much nicer way to go about this, thanks for the suggestion!

Reply:
Actually, turns out I cannot use the spread operator in this case, because the add function for a Set takes only one argument.

The following only adds '1' to the set:

```
const set = new Set();
const arr = [1, 2, 3, 4, 5];

set.add(...arr);
```

And it appears there are no alternative functions for adding a large number of items to the set.

Reply:
Just in case you were wondering where I was going with this, I think that you could have done something like:
` const phone_numbers = new Set([...c.responderPhoneNumbers, ...c.fallbackPhoneNumbers, ...c.heartbeatPhoneNumbers])`. But with the weird syntax it might have been less readable. Your way shows clearly what is going on.